frontend/catalyst/third_party/cuda/catalyst_to_cuda_interpreter.py

```diff
@@ -833,75 +822,42 @@ def cudaq_backend_info(device, _capabilities) -> BackendInfo:
             (QFunc, "extract_backend_info", cudaq_backend_info),
             (qml.QNode, "__call__", QFunc.__call__),
         ):
-            func = self.user_function
-            abs_axes = {}
-            static_args = None
-            # We could also pass abstract arguments here in *args
-            # the same way we do so in Catalyst.
-            # But I think that is redundant now given make_jaxpr2
-            jaxpr, _, out_treedef = trace_to_jaxpr(func, static_args, abs_axes, args, {})
+            jaxpr, out_type, treedef, dynamic_sig = super().capture(args)
 
-        # TODO(@erick-xanadu):
-        # What about static_args?
-        return jaxpr, out_treedef
+        # required because the CUDAQ interpretor requires host removed
+        jaxpr = remove_host_context(jaxpr)
+        jaxpr = jax._src.core.ClosedJaxpr(jaxpr, jaxpr.constvars)
 
+        return jaxpr, out_type, treedef, dynamic_sig
 
-def interpret(fun):
-    """Wrapper function that takes a function that will be interpretted with
-    the semantics in interpret_impl.
+    def jit_compile(self, args):
+        """
+        JIT compile needs to be overridden from the base class, for a couple of reasons:
 
-    Args:
-       fun: Function to be interpretted.
+        - It calls self.generate_ir() and self.compile(), none of which are applicable here
+          (for now).
+        - It tries to create a function cache and a CompiledFunction object, which will fail in
+          this case, because we have no compiled binary nor shared library.
+        """
 
-    Returns:
-       wrapped: A wrapped function that will do the tracing.
-    """
+        if self.jaxpr is None:
+            if self.user_sig and not self.compile_options.static_argnums:
+                msg = "Provided arguments did not match declared signature, recompiling..."
+                warnings.warn(msg, UserWarning)
+
+            self.jaxpr, _, self.out_treedef, self.c_sig = self.capture(args)
+
+        # Not entirely sure how to determine if 'type promotion' is needed, or if the default
```

Comment:
Is it a TODO? How is it handled in the base class?

Reply:
Most of this was written by Josh :sweat_smile: I don't want to spend too much time on this since it is not scheduled for this iteration, but I also didn't want it to not merge it. So I just fixed what was possible in a small amount of time.

Reply:
@rmoyard this is partly because I did not understand how this method works from the inhereted class docstring :) It returns True or False, but I wasn't understanding the implications.

Reply:
Up to you @erick-xanadu @josh146, I would either add a comment TODO here or match the behaviour of qjit.

Reply:
[078ad1f](https://github.com/PennyLaneAI/catalyst/pull/926/commits/078ad1f0f9e5e76db7178af8ce80e0052b847599)